File: Unit2-06/Airplane.ts
/**
 * This is the airplane class.
 *
 * By: Flynn Rundquist
 * Version: 1.0
 * Since: 2024-04-09
 */

class Airplane{
  public speed: number

constructor(speed: number = 0) {
  this.speed = speed
}

  // set speed
  public setSpeed(speed: number): number {
    this.speed = speed
    return this.speed
  }
}

export default Airplane

File: Unit2-06/Airplane.java
/**
 * Airplane class.
 *
 * @author Flynn
 * @version 1.0
 * @since 2024-04-24
 */

public class Airplane{
    private int speed;
    /**
     * This is the default constructor for the Airplane class.
     *
     * @param speed the speed
     */
    public Airplane(){
        speed = 0;
    }

    /**
     * This is the constructor for the Airplane class.
     *
     * @param speed the speed
     */
    public Airplane(int speed){
        this.speed = speed;
    }

    /**
     * This method sets the speed of the airplane.
     *
     * @param speed the speed
     * @return the speed
     */
    public int setSpeed(int speed){
        this.speed = speed;
        return speed;
    }
}
